### PR TITLE
muppet changes

### PR DESCRIPTION
**Pull request by jlevon**
NB: these have all been tested together, but I'm still planning to push as three separate commits as seen here. The first set at least is fairly inter-dependent.
I've added various testing notes to the relevant tickets.

### CODE REVIEW

**Comment by trentm** [comment]
Note to self, this covers:

MANTA-3252 use haproxy SSL termination for loadbalancer
MANTA-4434 loadbalancer could use a custom log for haproxy
MANTA-4741 size haproxy via SAPI

**Comment by jlevon** [comment]
Also

MANTA-1424 SSL certificate change should update loadbalancer's haproxy
MANTA-3526 want to disable tls 1.0
MANTA-4615 front door load balancing for buckets

:)

File: docs/index.md [hunk 1]
```diff
@@ -13,192 +13,80 @@ apisections:
...
-        "zookeeper": {
-            "servers": [ {
-                "host": "10.2.201.66",
-                "port": 2181
-            } ],
-            "timeout": 1000
-        }
-    }
+There is an `haproxy.cfg.in` template file; this is used by `muppet` to generate
+a new `haproxy.cfg` each time the topology of online loadbalancers changes.
```
[the remark this answers is not in the document]
Comment (jlevon) [reply docs/index.md]: oops, thanks

File: docs/index.md [hunk 2]
```diff
@@ -13,192 +13,80 @@ apisections:
...
+talking to the `haproxy` management interface; if not, then we refresh
+`haproxy`, which starts a new child process without interrupting existing
+connections.
+
+Note that `haproxy` itself is configured to do basic health checks on the
+backend servers, and will retire use of any unhealthy servers.
+
+## SAPI configuration
+
+The metadata key `SSL\_CERTIFICATE` of the loadbalancer SAPI service should
```
[the remark this answers is not in the document]
Comment (jlevon) [reply docs/index.md]: The backslash is incorrect, I had bad vim syntax highlighting...
Used in sapi_manifests/ssl_cert/template, writes the pem file out

File: docs/index.md [hunk 3]
```diff
@@ -13,192 +13,80 @@ apisections:
...
+contain the text of the private key used as the `haproxy` certificate; see
+above.
+
+The metadata key `HAPROXY\_NBTHREAD` defines the number of `haproxy` worker
```
[the remark this answers is not in the document]
Comment (jlevon) [reply docs/index.md]: will fix

File: docs/index.md [hunk 4]
```diff
@@ -13,192 +13,80 @@ apisections:
...
+live `/opt/smartdc/muppet/etc/haproxy.cfg` may be over-written by `muppet` at
+will.
+
+`haproxy` is configured to handle SSL termination. The certificate used is
+managed by `config-agent`, and can be updated in the Manta deployment zone using
+[manta-replace-cert](https://github.com/joyent/sdc-manta/blob/master/cmd/manta-replace-cert.js).
+This updates the SAPI metadata key `SSL_CERTIFICATE`, and be used without
+interrupting loadbalancer service.
+
+`haproxy` is set up to log (via syslog) to `/var/log/haproxy.log` in a format
```
[the remark this answers is not in the document]
Comment (jlevon) [reply docs/index.md]: That's correct. The default for both is 1024, which I felt we could reasonably bump up without any follow on issues. But we have to pick *something* so I picked that.

File: test/haproxy.cfg.in [hunk 5]
```diff
@@ -1,30 +1,35 @@
-# same as etc/haproxy.cfg.in, but without the errorfile directives, which break
-# unprivileged 'make test'
+# same as etc/haproxy.cfg.in, except able to use for unit tests
```
[the remark this answers is not in the document]
Comment (jlevon) [reply test/haproxy.cfg.in]: Yeah, that's a better idea.

**Comment by jlevon** [comment]
Hi, apologies, but I had to force push to fix up the history as I'm planning on pushing these three commits without squashing. There are no code changes to what you reviewed.

**Comment by jlevon** [comment]
NB since it turns out I can't rebase and merge anyway, I squashed them all.

**Pull request merged**